docs/understand/cmake_packages.rst

```diff
@@ -1,139 +1,188 @@
-===========================
+***********
 Using CMake
-===========================
+***********
 
-Most components in ROCm support CMake 3.5 or higher out-of-the-box and do not
-require any special Find modules. A Find module is often used by downstream to
-find the files by guessing locations of files with platform-specific hints.
-Typically, the Find module is required when the upstream is not built with CMake
-or the package configuration files are not available.
+Most components in ROCm support CMake. Projects depnding on header-only or
+library components typically require CMake 3.5 or higher whereas those wanting
+to make use of CMake's HIP language support will require CMake 3.21 or higher.
+
+Finding Dependencies
+====================
+
+.. note::
+   For a complete
+   reference on how to deal with dependencies in CMake, refer to the CMake docs
+   on `find_package
+   <https://cmake.org/cmake/help/latest/command/find_package.html>`_ and the
+   `Using Dependencies Guide
+   <https://cmake.org/cmake/help/latest/guide/using-dependencies/index.html>`_
+   to get an overview of CMake's related facilities.
+
+In short, CMake supports finding dependencies in two ways:
+
+-  In Module mode, it consults a file ``Find<PackageName>.cmake`` which tries to
+   find the component in typical install locations and layouts. CMake ships a
+   few dozen such scripts, but users and projects may ship them as well.
+-  In Config mode, it locates a file named ``<packagename>-config.cmake`` or
+   ``<PackageName>Config.cmake`` which describes the installed component in all
+   regards needed to consume it.
+
+ROCm dominantly relies on Config mode, one notable exception being the Module
+driving the compilation of HIP programs on Nvidia runtimes. As such, when
+dependencies are not found in standard system locations, one either has to
+instruct CMake to search for package config files in additional folders using
+the ``CMAKE_PREFIX_PATH`` variable (a semi-colon separated list of filesystem
+paths), or using ``<PackageName>_ROOT`` variable on a project-specific basis.
+
+There are nearly a dozen ways to set these variables. One may be more convenient
+over the other depending on your workflow. Conceptually the simplest is adding
+it to your CMake configuration command on the command-line via
+``-D CMAKE_PREFIX_PATH=....`` . AMD packaged ROCm installs can typically be
+added to the config file search paths such as:
+
+-  Windows: ``-D CMAKE_PREFIX_PATH=${env:HIP_PATH}``
+
+-  Linux: ``-D CMAKE_PREFIX_PATH=/opt/rocm``
 
 ROCm provides the respective *config-file* packages, and this enables
 ``find_package`` to be used directly. ROCm does not require any Find module as
 the *config-file* packages are shipped with the upstream projects.
```

Comment:
What are "the upstream projects" in this context?

Reply:
rocPRIM, hipBLAS, etc. Should we clarify?

Reply:
Yeah, adding an "e.g., rocPRIM or other ROCm libraries" would clear it up.

Reply:
Clarified.